wezterm-client/src/domain.rs

```diff
@@ -128,6 +129,14 @@ impl ClientDomainConfig {
         }
     }
 
+    pub fn local_echo_threshold_ms(&self) -> Option<u64> {
+        match self {
+            ClientDomainConfig::Unix(unix) => unix.local_echo_threshold_ms,
```

Comment:
we don't use predictions for local connections (the `!self.client.is_local()` logic in `should_predict` prior to this PR), so we should set it to `None` and remove the option from the unix domain as well, to preserve those semantics:

```suggestion
            ClientDomainConfig::Unix(unix) => None,
```

Reply:
Yup, but I've set the default to `None` for unix connections -- see `config/src/unix.rs` so the semantics should be the same by default. For the record, I found that allowing users to specify it for unix domains as well if they really want to might actually be a good idea. Following the test methodology explained in https://lwn.net/Articles/751763/ (that is, use the `typometer` benchmark from https://pavelfatin.com/typing-with-pleasure/), I ran a few small tests on my wezterm setup to measure typing latency. I'm using a Lenovo Carbon X1 with Debian / X / Gnome, and got the following results:

- Unix domain (very similar to a normal/native instance) w/o predictive echo: min=43.0ms, max=65.9ms, avg=57.1ms, sd=6.9ms

- Unix domain with predictive echo (threshold set to 0): min=25.5ms, max=49.1, avg=40.0ms, sd=7.5ms

 And that seems to be very reproducible. So, I figured offering the option might be nice given that it's not that complicated, but no strong opinion if you think it's not worth it.

Reply:
You make a good case for keeping that in, so let's do that :)